frontend/simple/views/containers/contributions/IncomeForm.vue

```diff
@@ -0,0 +1,245 @@
+    GroupPledgesGraph
+  },
+  props: {
+    isEditing: Boolean
+  },
+  data () {
+    return {
+      ephemeral: {
+        formVerified: false
+      },
+      form: {
+        option: null,
+        income: null,
+        pledge: null
+      },
+      // -- Hardcoded Data just for layout purposes:
+      fakeStore: {
+        mincome: 500,
+        mincomeFormatted: '$500'
+      }
+    }
+  },
+  computed: {
+    userPledgeAmount () {
+      return this.hasMinIncome & !!this.form.pledge ? Number(this.form.pledge) : null
+    },
+    userIncomeAmount () {
+      return this.hasLessIncome & !!this.form.income ? Number(this.form.income) : null
+    },
+    maxIncome () {
+      return this.fakeStore.mincome - 1
+    },
+    infoRequired () {
+      return this.L('This information is required.')
+    },
+    hasLessIncome () {
+      return this.$v.form.option.$model === 'no'
+    },
+    hasMinIncome () {
+      return this.$v.form.option.$model === 'yes'
+    },
+    hasLessIncomeError () {
+      if (!this.ephemeral.formVerified && this.$v.form.income.maxValue) {
+        return null
+      }
+
+      if (!this.$v.form.income.requiredIf) {
+        return this.infoRequired
+      } else if (!this.$v.form.income.maxValue) {
+        return this.L('If your income is the same or higher than the group\'s mincome change your answer to "Yes, I do"')
+      }
+    },
+    hasMinIncomeError () {
+      return this.ephemeral.formVerified && this.$v.form.pledge.$invalid ? this.infoRequired : null
+    },
+    saveButtonText () {
+      const verb = this.isFirstTime ? 'Add' : 'Save'
+
+      if (!this.$v.form.option.$model) { return this.L('{verb} details', { verb }) }
+      if (this.hasMinIncome) { return this.L('{verb} pledged details', { verb }) }
+      if (this.hasLessIncome) { return this.L('{verb} income details', { verb }) }
+    }
+  },
+  methods: {
+    verifyInputIncome: debounce(function (e) {
+      this.form.income = e.target.value
+      this.$v.form.income.$touch()
+    }, 0),
+    verifyInputPledge: debounce(function (e) {
+      this.form.pledge = e.target.value
+      this.$v.form.pledge.$touch()
+    }, 0),
+    verifyForm () {
+      this.ephemeral.formVerified = true
+
+      if (!this.$v.form.$invalid) {
+        const makeIncome = this.$v.form.option.$model === 'yes'
+
+        this.$emit('save', {
+          makeIncome,
+          amount: makeIncome ? this.$v.form.pledge.$model : this.$v.form.income.$model
+        })
+      }
+    },
+    cancelForm () {
+      this.$emit('cancel')
+    },
+    resetFormVerify () {
+      this.ephemeral.formVerified = false
+    }
+  },
+  validations: {
+    form: {
+      option: {
+        required
+      },
+      income: {
+        requiredIf: requiredIf(model => model.option === 'no'),
+        // REVIEW - how to do this value dynamic?! this.mincome doesn't work
+        maxValue: maxValue(499)
+      },
+      pledge: {
+        requiredIf: requiredIf(model => model.option === 'yes')
```

Comment:
Also, both here and on the income validator, it doesn't seem to me like the validator here is being written correctly because of how it accesses `model.option`. According to the vuelidate docs example, the passed in `model` parameter represents the `pledge` parameter (if it were a nested parameter). Since it's not a nested parameter, there is no `pledge.option` value here. Similarly, there is no `income.option` value on line 235 above, because `income` is not a nested field.

You probably want `model === 'yes'` here, I'm guessing. Not sure. You can also make it even clearer by checking `this.form.pledge`.

Also, I would move the validations section next to the data section (above the methods) so that it's easier to see what values they are validating.

Reply:
Fixed, please verify if it's ok